addons/website_sale/models/product_template.py

```diff
@@ -471,6 +475,7 @@ def _get_additionnal_combination_info(self, product_or_template, quantity, date,
         if pricelist_rule_id:  # If a rule was applied, there might be a discount
             # For ecommerce flows, the base price is always the product sales price
             # which can be computed by calling `_compute_base_price` without a pricelist rule
+            pricelist_item = self.env['product.pricelist.item'].browse(pricelist_rule_id)
```

Comment:
should initialize `pricelist_item` variable before if to avoid error when there in no `pricelist_rule_id`

pricelist_item = self.env['product.pricelist.item']